### DISCUSSION TITLE
Is there a utility available to fill in auctions for PBN files?

### DISCUSSION THREAD

**Discussion by Rick-Wilson** [lorserker/ben#81] [Q&A] [answered]
Not at all sure if this is the right place to ask this . . .

I have a lot of (free-to-use) script-generated PBN files, for use as practice hands.  These are for my students and anyone else who wants them.  They are organized by convention, or bidding scenario.  They can be used in BBO or other online platforms.

I would like to have a utility where I can feed a each PBN file to a bot, and it will create a 2/1 auction and save it back to the PBN file.  I have maybe 200 such files, many with 100 hands each, so I'd like something automated.  With an embedded auction, I have scripts to produce companion PDFs of the hands, so students can check their own auctions after each hand.

I know this AI can do bidding.  I am wondering if there's a utility already made to interface it with PBN files.  Or if not, is it a reasonable application of the software.

I've attached a sample PBN file here.
[1.10 Jacoby 2NT - Leveled x100 - NS.pdf](https://github.com/lorserker/ben/files/13507454/1.10.Jacoby.2NT.-.Leveled.x100.-.NS.pdf)
[1.10 Jacoby 2NT - Leveled x100 - NS.pbn.zip](https://github.com/lorserker/ben/files/13507464/1.10.Jacoby.2NT.-.Leveled.x100.-.NS.pbn.zip)

**Comment by ThorvaldAagaard** [comment, marked as answer]
Hi Rick
First of all I would like a copy of your files, and use it for training BEN :-)

Our procedure at the moment is as follows:
1: Gather interesting boards matching normal bridge problems 
2: Ask Bridge Bidding Analyzer to bid the hands according to the CC defined (https://sites.google.com/view/bbaenglish)
3: Use the generated bidding from BBA to train BEN

So you are looking for ttps://sites.google.com/view/bbaenglish that can bid the hands according to 2/1-system

I have used it for bidding your sample PBN-file (just remove the extension .txt that github forces me to have)
[BBA - 30.11.2023 13.23.pbn.txt](https://github.com/lorserker/ben/files/13512694/BBA.-.30.11.2023.13.23.pbn.txt)
It will bid all the hands, but as a teacher you should probably look thru the deals to check if it is as expected - I noticed  this, that could be a problem:
![image](https://github.com/lorserker/ben/assets/10560652/5a44ae06-c7d2-4cf1-9bac-4622150f4492)
as I am not sure Bergen is on for your students


Now I can use this file for training BEN, and your students would be able to bid with BEN (is that is interesting) but it will require some work, and you would have to run BEN from some server.

But please send me the files at thorvald@netbridge.dk

I can also recommend you look at https://github.com/ADavidBailey/Practice-Bidding-Scenarios/blob/main/Documents/Better%20Bidding%20Practice.md for online bidding practice
